source/spark_sql_migrations/tests/test_readme.py

```diff
@@ -0,0 +1,61 @@
+import re
+import os
+import tests.helpers.mock_helper as mock_helper
+from unittest.mock import Mock
+import spark_sql_migrations.container as container
+import spark_sql_migrations.schema_migration_pipeline as schema_migration_pipeline
+
+
+def _get_source_path() -> str:
+    working_directory = os.getcwd()
+    source_path = re.match(r"(.+?source)", working_directory).group(1)
+    return source_path
+
+
+def _check_code_syntax(code: str):
+    try:
+        byte_code = compile(code, "<string/>", "exec")
+        exec(byte_code)
+    except Exception as e:
+        raise e
+    return "Syntax is correct."
+
+
+def test_read_me_python_code_example(mocker: Mock) -> None:
+    # Arrange
+    mocker.patch.object(
+        container,
+        container.create_and_configure_container.__name__,
+        side_effect=mock_helper.do_nothing
+    )
+
+    mocker.patch.object(
+        schema_migration_pipeline,
+        schema_migration_pipeline.migrate.__name__,
+        side_effect=mock_helper.do_nothing
+    )
+
+    source_path = _get_source_path()
+    readme_path = f"{source_path}/spark_sql_migrations/README.md"
+
+    # Act
+    with open(readme_path, "r") as file:
+        lines = file.readlines()
+        code_block = ""
+        in_code_block = False
+
+        for line in lines:
```

Comment:
nit: I assume this works as intended. But if you're interested you could look into segregating the parsing from the assertions. I think that a finite-state machine would shine here. The book "The Pragmatic Programmer" has a nice chapter about the concept.

Reply:
Tbh. the code is created from Github co-pilot with almost no changes by me. I have now refactored the test and a method has been created, which we could re-use in another test if needed :)